new/prusti-specs/src/specifications/untyped.rs

```diff
@@ -169,6 +170,9 @@ impl AssignExpressionId<Assertion> for common::Assertion<(), syn::Expr, Arg> {
 /// A trait for encoding the statements for type-checking the spec.
 pub trait EncodeTypeCheck {
     fn encode_type_check(&self, tokens: &mut TokenStream);
+    fn encode_type_check_any_type(&self, tokens: &mut TokenStream) {
+        unimplemented!();
```

Comment:
`unimplemented!` means that something is not yet implemented. Is this really the case, or should it be `unreachable!`?